code/go/internal/validator/semantic/validate_minimum_kibana_version.go

```diff
@@ -0,0 +1,153 @@
+
+	ve "github.com/elastic/package-spec/v2/code/go/internal/errors"
+	"github.com/elastic/package-spec/v2/code/go/internal/fspath"
+	"github.com/elastic/package-spec/v2/code/go/internal/pkgpath"
+)
+
+// ValidateMinimumKibanaVersion if the package is an input package, and the package version is >= 1.0.0,
+// then the kibana version condition must be >= 8.8.0
+func ValidateMinimumKibanaVersion(fsys fspath.FS) ve.ValidationErrors {
+	manifest, err := readManifest(fsys)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	packageType, err := getPackageType(*manifest)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	packageVersion, err := getPackageVersion(*manifest)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	kibanaVersionCondition, err := getKibanaVersionCondition(*manifest)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	err = validateMinimumKibanaVersion(packageType, *packageVersion, kibanaVersionCondition)
+	if err != nil {
+		return ve.ValidationErrors{err}
+	}
+
+	return nil
+}
+
+func validateMinimumKibanaVersion(packageType string, packageVersion semver.Version, kibanaVersionCondition string) error {
+
+	if packageType != "input" {
+		return nil
+	}
+
+	if packageVersion.LessThan(semver.MustParse("1.0.0")) {
+		return nil
+	}
+
+	if kibanaVersionConditionIsGreaterThanOrEqualTo8_8_0(kibanaVersionCondition) {
+		return nil
+	}
+
+	return errors.New("Warning: conditions.kibana.version must be ^8.8.0 or greater for non experimental input packages (version > 1.0.0)")
+}
+
+func readManifest(fsys fspath.FS) (*pkgpath.File, error) {
+	manifestPath := "manifest.yml"
+	f, err := pkgpath.Files(fsys, manifestPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "can't locate manifest file")
+	}
+
+	if len(f) != 1 {
+		return nil, errors.New("single manifest file expected")
+	}
+
+	return &f[0], nil
+}
+
+func getPackageType(manifest pkgpath.File) (string, error) {
+
+	val, err := manifest.Values("$.type")
+	if err != nil {
+		return "", errors.Wrap(err, "can't read manifest type")
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return "", errors.New("manifest type is not a string")
+	}
+
+	return sVal, nil
+}
+
+func getPackageVersion(manifest pkgpath.File) (*semver.Version, error) {
+
+	val, err := manifest.Values("$.version")
+	if err != nil {
+		return nil, errors.Wrap(err, "can't read manifest version")
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return nil, errors.New("package version is not a string")
+	}
+
+	sVersion, err := semver.NewVersion(sVal)
+	return sVersion, nil
+}
+
+func getKibanaVersionCondition(manifest pkgpath.File) (string, error) {
+
+	val, err := manifest.Values("$.conditions[\"kibana.version\"]")
+	if err != nil {
+		return "", nil
+	}
+
+	sVal, ok := val.(string)
+	if !ok {
+		return "", errors.New("manifest kibana version is not a string")
+	}
+
+	return sVal, nil
+}
+
+func kibanaVersionConditionIsGreaterThanOrEqualTo8_8_0(kibanaVersionCondition string) bool {
```

Comment:
@jsoriano I am not sure if this function is a bit too hacky. I am a bit confused about the spec for `conditions.kibana.version`. All of the integrations use `^` and `||` syntax. However in the spec we say that `>=` can be used https://github.com/elastic/package-spec/blob/57e322febbb4b22e72b60326d3e64955a2d9d7b5/spec/integration/manifest.spec.yml#L113.

Reply:
Yes, in principle any comparison is allowed here as supported by the underlying library https://github.com/Masterminds/semver/#checking-version-constraints

Your implementation makes sense to me. I guess it will only have problems if someone uses "less than" conditions, like `< 8.10.0`, but I don't think they make a lot of sense in this context.